src/main/kotlin/com/sourcegraph/cody/config/actions/OpenCodySettingsEditorAction.kt

```diff
@@ -0,0 +1,72 @@
+package com.sourcegraph.cody.config.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.util.io.write
+import com.jetbrains.jsonSchema.JsonSchemaMappingsProjectConfiguration
+import com.jetbrains.jsonSchema.UserDefinedJsonSchemaConfiguration
+import com.jetbrains.jsonSchema.ide.JsonSchemaService
+import com.jetbrains.jsonSchema.impl.JsonSchemaVersion
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.common.ui.DumbAwareEDTAction
+import com.sourcegraph.config.ConfigUtil
+import com.sourcegraph.utils.CodyEditorUtil
+import kotlin.io.path.name
+
+class OpenCodySettingsEditorAction : DumbAwareEDTAction("Open Cody Settings Editor") {
+  override fun actionPerformed(e: AnActionEvent) {
+    val project =
+        e.project
+            ?: run {
+              logger.warn("Project cannot be null")
+              return
+            }
+
+    val settingsVf =
+        CodyEditorUtil.createFileOrScratchFromUntitled(
+            project, ConfigUtil.getSettingsFile(project).toString(), content = "{\n  \n}")
+            ?: run {
+              logger.warn("Could not create settings file")
+              return
+            }
+
+    CodyEditorUtil.showDocument(project, settingsVf)
+
+    reloadSchemaAsync(project)
+  }
+
+  private fun reloadSchemaAsync(project: Project) {
+    CodyAgentService.withAgentRestartIfNeeded(project) { agent ->
+      val settingsSchema = agent.server.extensionConfiguration_getSettingsSchema(null).get()
```

Comment:
so this actions not only opens the file but also fetches the config from the agent and updates the content of the file, is that right? that makes the agent the source of truth at the moment we use this action - I wonder, shouldn't it be the client holding the truth?

Reply:
btw, `extensionConfiguration_getSettingsSchema` always takes `null`, why is that? 🤔

Reply:
>  I wonder, shouldn't it be the client holding the truth?
I do not think so, and I'm not sure how that would be supposed to work?
After all it's VSC code which knows which settings are used, etc.
Keeping it in client would be artificial, and would also need to be duplicated for every client.

Reply:
Ok. My main concern was about persisting the settings b/w the IDE runs. Now I can see that `ConfigUtil::getAgentConfiguration` handles the initialization properly on the startup - all good 👍